### ISSUE TITLE
Formating broken by message(NULL, appendLF = FALSE) 

### ISSUE THREAD

**Issue by courtiol** [rstudio/rmarkdown#1849]
Hi all,
It seems that printing an empty message with the option ` appendLF = FALSE` is sufficient to break the formatting:

````
---
output: html_document
---

```{r}
message(NULL, appendLF = FALSE) ## an empty message
```

```{r}
1+1
```
````
Here is what it leads to:
<img width="444" alt="Screen Shot 2020-06-23 at 09 46 27" src="https://user-images.githubusercontent.com/7499666/85375329-6d9ba600-b536-11ea-8d30-64c9314b5725.png">

My session info:
``` r
xfun::session_info("rmarkdown")
#> R version 4.0.0 (2020-04-24)
#> Platform: x86_64-apple-darwin17.0 (64-bit)
#> Running under: macOS Catalina 10.15.5
#> 
#> Locale: en_US.UTF-8 / en_US.UTF-8 / en_US.UTF-8 / C / en_US.UTF-8 / en_US.UTF-8
#> 
#> Package version:
#>   base64enc_0.1.3 digest_0.6.25   evaluate_0.14   glue_1.4.1     
#>   graphics_4.0.0  grDevices_4.0.0 highr_0.8       htmltools_0.5.0
#>   jsonlite_1.6.1  knitr_1.29      magrittr_1.5    markdown_1.1   
#>   methods_4.0.0   mime_0.9        rlang_0.4.6     rmarkdown_2.3.1
#>   stats_4.0.0     stringi_1.4.6   stringr_1.4.0   tinytex_0.24   
#>   tools_4.0.0     utils_4.0.0     xfun_0.15       yaml_2.2.1     
#> 
#> Pandoc version: 2.9.2.1
```

**Comment by cderv** [comment]
This would be a behavior from **knitr** directly 

``` r
cat(knitr::knit(text = c("```{r, echo = FALSE}", "message(NULL, appendLF = FALSE)", "```"), quiet = TRUE))
#> 
#> ```
#> #> ```
```

When empty, **knitr** does not add a newline. 

``` r
cat(knitr::knit(text = c("```{r, echo = FALSE}", "message('a', appendLF = FALSE)", "```"), quiet = TRUE))
#> 
#> ```
#> #> a
#> ```
```

When not empty, a newline will be appended. 

This comes from internal  `knitr::comment_out()` function. By default `newline = TRUE` in its argument, but it does not work with empty string.

``` r
knitr:::comment_out("")
#> [1] "## "
knitr:::comment_out("a")
#> [1] "## a\n"
```

@courtiol I know it has been a while but is there a real use case where this happened (empty message ?) ?

thanks.

**Comment by courtiol** [comment]
@cderv, there was a use case but I am afraid I can no longer remember the specifics.

I think I had use `message()` to output something, I had not realized that this something could sometimes be NULL, and I struggled a little to understand the source of the problem since the vignette would no longer compile.

I could attempt Rcheology if very necessary, but I think that the reason I did not describe the context is because it was quite complex (I am using function to compile additional vignette on the fly in some packages).
